app/code/community/Jowens/JobQueue/Model/Worker.php

```diff
@@ -65,8 +65,18 @@ public function executeJobs($schedule=null) {
             ->addFieldToFilter('failed_at', array('null' => true))
             ->addFieldToFilter('attempts', array('lt' => (int)Mage::getStoreConfig('jobqueue/config/max_attempts')));
 
-            // randomly order to prevent lock contention among workers
-            $collection->getSelect()->order(new Zend_Db_Expr('RAND()'));
+            if (Mage::getStoreConfigFlag('jobqueue/config/sort_random')) {
+                // randomly order to prevent lock contention among workers
+                $collection->getSelect()->order(new Zend_Db_Expr('RAND()'));

```

Comment:
I don't know why this was added, it doesn't really have any benefit. Can you just remove and sort the collection by ascending ID?

Reply:
Well the comment says: `randomly order to prevent lock contention among workers`. With the added system config setting it's now configurable. That's more backwards compatible then just removing it.